sentry/src/routes/campaign.rs

```diff
@@ -1249,4 +1299,82 @@ mod test {
             assert_eq!(delta_budget, Some(DeltaBudget::Decrease(decrease_by)));
         }
     }
+
+    #[tokio::test]
+    async fn campaign_is_closed_properly() {
+        let multiplier = 10_u64.pow(UnifiedNum::PRECISION.into());
+        let campaign = DUMMY_CAMPAIGN.clone();
+
+        let app = setup_dummy_app().await;
+
+        insert_channel(&app.pool, campaign.channel.clone())
+            .await
+            .expect("Should insert dummy channel");
+        insert_campaign(&app.pool, &campaign)
+            .await
+            .expect("Should insert dummy campaign");
+
+        // Test if remaining is set to 0
+        {
+            let remaining = UnifiedNum::from_u64(100 * multiplier);
```

Comment:
I would suggest to use the `campaign.budget` for this value. If we change it to something `< 100` tests will randomly start failing